homeassistant/components/history_stats/sensor.py

```diff
@@ -108,6 +108,7 @@ def __init__(
         self, hass, entity_id, entity_states, start, end, duration, sensor_type, name
     ):
         """Initialize the HistoryStats sensor."""
+        self.hass = hass
```

Comment:
Let's remove this asap when the tests are refactored. It gives the wrong impression about how the entity should be used.

Reply:
Agreed.  I'll refactor these once I have had some sleep.

Reply:
Cleaned up the moved tests here https://github.com/home-assistant/core/pull/49446 and removed `self.hass`